### PR TITLE
[12.0][MIG] website_sale_checkout_country_vat

### PR DESCRIPTION
**Pull request by Tardo**
Normal migration... Odoo 12.0 doesn't have "vat" as required field.

### CODE REVIEW

**Comment by Tardo** [comment]
Runbot fails because missing dependencies.... (are in other repo and not merged into 12.0)

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Trying on runbot, I find these problems:

- When not logged in, pressing "Sign up and checkout" leads to `404` page
- When logged in by other means, if I edit the address, I get `500: Internal Server Error`.

**Comment by pedrobaeza** [comment]
About Travis tests, can be due to this module.

**Comment by pedrobaeza** [comment]
I still get 404 page when I click in "Sign up and checkout". Is it due to this module? I have been able however to see the country dropdown when editing the address once logged-in.

**Comment by Tardo** [comment]
Dropped oca_dependencies changes.

File: website_sale_checkout_country_vat/static/src/js/dropdown.js [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
-        selector: ".oe_website_sale:has(.js_country_dropdown, \
-                                        select[name=country_id])",
+    var CheckoutCountryVatDropdown = animation.Class.extend({
+        selector: ".oe_website_sale:has(.js_country_dropdown, " +
+                  "select[name=country_id])",
         start: function (editable_mode) {
             var result = this._super(editable_mode);
-            this.$address_country = this.$("select[name=country_id]");
-            this.$vat_no_country_field = this.$("#no_country_field");
+            this.$address_country = this.$target.find(
```
Comment (yajo) [inline website_sale_checkout_country_vat/static/src/js/dropdown.js]: I'm curious: why are you removing the `this.$` shorthand? It's available in all subclasses of `Widget`...

**Review by yajo — CHANGES_REQUESTED** [review]

File: website_sale_checkout_country_vat/static/src/js/dropdown.js (hunk 1 above)
Comment (Tardo) [reply website_sale_checkout_country_vat/static/src/js/dropdown.js]: Umm to ensure that doesn't have conflicts if the snippet it's used more than one time in the same page...

Comment (yajo) [reply website_sale_checkout_country_vat/static/src/js/dropdown.js]: Don't worry, it doesn't happen.

If you were to use `$(selector)`, yes, it's raw jquery and selects in the whole page, but `this.$(selector)` is different and only searches inside the widget.

**Comment by Tardo** [comment]
Reverted "$target" changes

**Comment by Tardo** [comment]
@pedrobaeza The problem is that "website_sale_suggest_create_account" module  fails on runbot... I installed it on local and works :/

**Comment by pedrobaeza** [comment]
Please check with @Yajo what can be the problem.

**Comment by pedrobaeza** [comment]
Although I'm now seeing that this PR is green.

**Review by pedrobaeza — APPROVED** [review]

**Review by yajo — APPROVED** [review]

**Pull request merged**